### PR TITLE
AWS Docs

### PR DESCRIPTION
**Pull request by datosh**
Signed-off-by: Fabian Kammel <fk@edgeless.systems>

### Proposed change(s)

- Install docs
- Getting started

### Checklist

- [ ] Update [CHANGELOG.md](https://github.com/edgelesssys/constellation/blob/main/CHANGELOG.md)
- [ ] Update [docs](https://github.com/edgelesssys/constellation/tree/main/docs)
- [ ] Link to Milestone

### CODE REVIEW

File: docs/docs/getting-started/first-steps.md [hunk 1]
```diff
@@ -164,6 +164,35 @@ If you don't have a cloud subscription, check out [MiniConstellation](first-step
 
         Supported are all machines from the N2D family. It defaults to `n2d-standard-4` (4 vCPUs, 16 GB RAM), but you can use any other VMs from the same family. Refer to [N2D machine series](https://cloud.google.com/compute/docs/general-purpose-machines#n2d_machines) or run `constellation config instance-types` to get the list of all supported options.
 
+    </tabItem>
+    <tabItem value="aws" label="AWS">
+
+    * **region**: The name of your chosen AWS data center region, e.g., `us-east-2`.
+
+        TODO: What is the list of supported regions for our images?
```
Comment (datosh) [inline docs/docs/getting-started/first-steps.md:172]: @malt3 could you help me out with that point? `config generate` does not fill that value in. I see that this is not a static value as with GCP or Azure, should we add some logic to CLI to fetch this value based on the region? Is a manual approach currently preferred?

Comment (datosh) [inline docs/docs/getting-started/first-steps.md:172]: @malt3 Is [this](https://github.com/edgelesssys/constellation/blob/feat/docs/aws-install/.github/workflows/build-os-image.yml#L270-L271) the full list of regions we target for release?

File: docs/docs/getting-started/first-steps.md [hunk 2]
```diff
@@ -164,6 +164,35 @@ If you don't have a cloud subscription, check out [MiniConstellation](first-step
...
+
+        You can find a list of all [available regions in AWS' documentation](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-regions-availability-zones.html#concepts-available-regions).
+
+    * **zone**: The name of your chosen AWS data center availability zone, e.g., `us-east-2a`.
+
+        Learn more about [availability zones in AWS' documentation](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-regions-availability-zones.html#concepts-availability-zones).
+
+    * **image**: TODO: How do customers find these?
```
[the remark this answers is not in the document]
Comment (3u13r) [reply docs/docs/getting-started/first-steps.md:180]: Until https://github.com/edgelesssys/constellation/pull/416 is decided on and implemented, I think we maybe just add the mapping from region to ami as a comment behind the field by hand.

File: docs/docs/getting-started/first-steps.md (hunk 1 above)
Comment (3u13r) [reply docs/docs/getting-started/first-steps.md:172]: Yes, we decided on having one image in every major region. Of course images from us-east-2 cannot be used in us-east-1, but I hope our users are flexible here. Supporting more location will be a pain, since we probably want to clean up old images at some point and we would have to do that in every region then.

Comment (malt3) [reply docs/docs/getting-started/first-steps.md:172]: We should add a disclaimer here (and maybe in the code if we see no AMI for the region) that users can open an issue to get the image replicated into their region.

Comment (malt3) [reply docs/docs/getting-started/first-steps.md:172]: I can create an issue template that we can link to for that purpose

File: docs/docs/getting-started/first-steps.md [hunk 3]
```diff
@@ -179,6 +208,8 @@ If you don't have a cloud subscription, check out [MiniConstellation](first-step
     constellation config fetch-measurements
     ```
 
+    TODO: Is this supported for AWS? Or do we need a hint?
```
Comment (malt3) [inline docs/docs/getting-started/first-steps.md:211]: The upload to S3 has to happen manually for now using each AMI ID. The download should work (I think). This will be fully supported by #416

File: docs/docs/workflows/scale.md [hunk 4]
```diff
@@ -63,6 +63,11 @@ Alternatively, you can manually scale your cluster up or down:
 2. **Edit** the **worker** instance group.
 3. Set the new **number of instances** and **save**.
 
+</tabItem>
+<tabItem value="aws" label="AWS">
+
+TODO: Is this already supported? Test or write note here!
```
Comment (malt3) [inline docs/docs/workflows/scale.md:69]: No yet supported. See AB#2441

File: docs/docs/workflows/storage.md [hunk 5]
```diff
@@ -36,6 +36,11 @@ This includes support for [volume snapshots](https://cloud.google.com/kubernetes
 You can use them to bring a volume back to a prior state or provision new volumes.
 Follow the instructions on how to [install the Constellation CSI driver](#installation) or check out the [repository](https://github.com/edgelesssys/constellation-gcp-compute-persistent-disk-csi-driver) for information about the configuration.
 
+</tabItem>
+<tabItem value="aws" label="AWS">
+
+TODO: Is this already supported? Test or write note here!
```
Comment (malt3) [inline docs/docs/workflows/storage.md:42]: @daniel-weisse should know more about this :)

File: docs/docs/workflows/troubleshooting.md [hunk 6]
```diff
@@ -35,5 +35,10 @@ Constellation uses the default bucket to store logs. Its [default retention peri
 
 :::
 
+</tabItem>
+<tabItem value="aws" label="AWS">
+
+TODO: Is this already supported? Test or write note here!
```
Comment (malt3) [inline docs/docs/workflows/troubleshooting.md:41]: No

Comment (datosh) [reply docs/docs/workflows/troubleshooting.md]: Was implemented, added missing permissions in #461 .

File: docs/docs/getting-started/first-steps.md (hunk 1 above)
Comment (datosh) [reply docs/docs/getting-started/first-steps.md]: I re-used the feature template with a prefilled title, feel free to improve this as you see fit :)

File: docs/docs/workflows/storage.md (hunk 5 above)
Comment (datosh) [reply docs/docs/workflows/storage.md]: Added disclaimer.

File: docs/docs/workflows/storage.md [hunk 7]
```diff
@@ -36,6 +36,17 @@ This includes support for [volume snapshots](https://cloud.google.com/kubernetes
...
 Follow the instructions on how to [install the Constellation CSI driver](#installation) or check out the [repository](https://github.com/edgelesssys/constellation-gcp-compute-persistent-disk-csi-driver) for information about the configuration.
 
+</tabItem>
+<tabItem value="aws" label="AWS">
+
+:::caution
+
+Storage isn't yet implemented for AWS. If you require this feature, [let us know](https://github.com/edgelesssys/constellation/issues/new?assignees=&labels=&template=feature_request.md)!
+
+Other CSI drivers that are compatible with Kubernetes on AWS can be used.
```
Comment (thomasten) [inline docs/docs/workflows/storage.md:46]:
```suggestion
Confidential storage isn't yet implemented for AWS. If you require this feature, [let us know](https://github.com/edgelesssys/constellation/issues/new?assignees=&labels=&template=feature_request.md)!

You may use other (non-confidential) CSI drivers that are compatible with Kubernetes on AWS.
```

File: docs/docs/workflows/recovery.md [hunk 8]
```diff
@@ -87,6 +87,15 @@ If this fails due to an unhealthy control plane, you will see log messages simil
 
 This means that you have to recover the node manually.
 
+</tabItem>
+<tabItem value="aws" label="AWS">
+
+:::caution
+
+Recovery isn't yet implemented for AWS. If you require this feature, [let us know](https://github.com/edgelesssys/constellation/issues/new?assignees=&labels=&template=feature_request.md)!
```
Comment (daniel-weisse) [inline docs/docs/workflows/recovery.md:95]: Is there actually anything blocking us from supporting recovery?
As far as I know, this should just require us to set up a loadbalancer the CLI talks tom the rest should be CSP independent.

File: docs/docs/getting-started/install.md [hunk 9]
```diff
@@ -132,6 +132,138 @@ You need the following permissions on this project:
...
+                "ec2:DeleteSecurityGroup",
+                "ec2:DeleteVpc"
+            ],
+            "Resource": "*"
+        }
+    ]
+}
+```
+
+Follow Amazons' guide on [understanding](https://docs.aws.amazon.com/IAM/latest/UserGuide/access_policies.html) and [managing policies](https://docs.aws.amazon.com/IAM/latest/UserGuide/access_policies_managed-vs-inline.html), and [roles](https://docs.aws.amazon.com/IAM/latest/UserGuide/access_iam-tags.html).
```
Comment (Nirusu) [inline docs/docs/getting-started/install.md:265]: In my opinion the whole sentence is a bit awkward to read.
"Follow Amazon's guide on understanding and managing policies" by itself sounds fine, though I think both articles are actually about understanding the purpose of certain kinds of policies.

"Follow Amazon's guide on understanding and managing policies_, and roles_" sounds off to me. The comma implies a break in speech, yet "and roles" is meant to be read as "Follow Amazon's guide on understanding [...] roles" or "Follow Amazon's guide on managing [...] roles".

Not sure which one, though. And having two "ands" in the sentence is also not good, which makes it a confusing enumeration.

I would rewrite this, though I don't have a concrete suggestion as I don't fully know the articles and context of the linked articles to make a suggestion.

File: docs/docs/workflows/scale.md [hunk 10]
```diff
@@ -88,6 +97,15 @@ To increase the number of control-plane nodes, follow these steps:
 2. **Edit** the **control-plane** instance group.
 3. Set the new (increased) **number of instances** and **save**.
 
+</tabItem>
+<tabItem value="aws" label="AWS">
+
+:::caution
+
+Scaling isn't yet implemented for AWS. If you require this feature, [let us know](https://github.com/edgelesssys/constellation/issues/new?assignees=&labels=&template=feature_request.md)!
```
Comment (Nirusu) [inline docs/docs/workflows/scale.md:105]: Why do we want users to create issues for everything we likely should support in the first place (and likely will add later anyways)? Or do we plan on abandoning AWS after the release of 2.2.0 until we get user feedback?

File: docs/docs/workflows/storage.md [hunk 11]
```diff
@@ -36,6 +36,17 @@ This includes support for [volume snapshots](https://cloud.google.com/kubernetes
 You can use them to bring a volume back to a prior state or provision new volumes.
 Follow the instructions on how to [install the Constellation CSI driver](#installation) or check out the [repository](https://github.com/edgelesssys/constellation-gcp-compute-persistent-disk-csi-driver) for information about the configuration.
 
+</tabItem>
+<tabItem value="aws" label="AWS">
+
+:::caution
+
+Storage isn't yet implemented for AWS. If you require this feature, [let us know](https://github.com/edgelesssys/constellation/issues/new?assignees=&labels=&template=feature_request.md)!
```
Comment (Nirusu) [inline docs/docs/workflows/storage.md:44]: Same here, why do we suggest users to create an issue?

File: docs/docs/workflows/scale.md (hunk 10 above)
Comment (Nirusu) [reply docs/docs/workflows/scale.md:105]: Ah, bad wording on my side, but I see.

I wasn't so sure about whether it was just not available since AWS was just recently implemented and therefore not everything is ready yet or if we actually deprioritize it / don't want to work on it for now as technically AWS also offers the weakest security model of the CSPs we support. 

An alternative suggestion I have is to reword this as _will become available in a later release_ but still suggest opening an issue if this is a priority for someone. I think that is more neutral and reflects that AWS is more in an early state.

But don't need to change it then if we actually don't want to focus on AWS for now until we know users actually want to use it. I think that's fine then.

File: docs/docs/getting-started/install.md [hunk 12]
```diff
@@ -132,6 +132,138 @@ You need the following permissions on this project:
 
 Follow Google's guide on [understanding](https://cloud.google.com/iam/docs/understanding-roles) and [assigning roles](https://cloud.google.com/iam/docs/granting-changing-revoking-access).
 
+</tabItem>
+<tabItem value="aws" label="AWS">
+
+In the following steps you will use Constellation to perform two tasks: create policies & infrastructure.
```
[the remark this answers is not in the document]
Comment (datosh) [reply docs/docs/getting-started/install.md:138]: I added more context to help with the motivation, @thomasten.

File: docs/docs/workflows/recovery.md (hunk 8 above)
Comment (datosh) [reply docs/docs/workflows/recovery.md:95]: @3u13r is working on the loadbalancer part. I can take another stab at recovery once it's merged.

File: docs/docs/workflows/scale.md (hunk 10 above)
Comment (datosh) [reply docs/docs/workflows/scale.md:105]: I hinted at the fact that we have those tasks in our backlog: "Scaling isn't **yet** implemented for AWS.", users should engage with us (via issues) if this is an urgent need.

I will resolve this conversation for now, feel free to reopen if you feel we should change/improve the wording.

File: docs/docs/getting-started/install.md (hunk 9 above)
Comment (datosh) [reply docs/docs/getting-started/install.md]: I agree! I have removed the last link since it was off-context. The first two resources should provide enough information to know how policies work.

**Comment by datosh** [comment]
Wait for #416 to be merged, and update AMI-related information before merging.

File: docs/docs/workflows/recovery.md (hunk 8 above)
Comment (daniel-weisse) [reply docs/docs/workflows/recovery.md:95]: Tested this with with a single control-plane cluster, and a 3 control-plane nodes cluster.
Both self recovery (at least half of the control-planes are still reachable) and manually recovery using the CLI work on AWS.

Comment (3u13r) [reply docs/docs/workflows/recovery.md:95]: We still need to manually pass the IP to `constellation recover`, right? So I see two ways forward:
1. Keep saying that `constellation recover` is not supported
2. Document the way of getting the right IP from AWS and recovering with that. 

I have no real opinion on which way we choose.

File: docs/docs/getting-started/first-steps.md [hunk 13]
```diff
@@ -164,6 +171,46 @@ If you don't have a cloud subscription, check out [MiniConstellation](first-step
...
+
+    * **iamProfileControlPlane**: The name of a profile attached to all control plane nodes.
+
+        Use the [provided Terraform script](https://github.com/edgelesssys/constellation/blob/main/hack/terraform/aws/iam) to generate the necessary profile. It will be provided as Terraform output variable: `control_plane_instance_profile`.
+
+        Alternatively, you can create the AWS profile with a tool of your choice. Use the JSON policy in [main.tf](https://github.com/edgelesssys/constellation/blob/main/hack/terraform/aws/iam/main.tf) in the resource `aws_iam_policy.control_plane_policy`.
+
+    * **iamProfileWorkerNodes**: The name of a profile attached to all control plane nodes.
+
+        Use the [provided Terraform script](https://github.com/edgelesssys/constellation/blob/main/hack/terraform/aws/iam) to generate the necessary profile. It will be provided as Terraform output variable: `worker_nodes_instance_profile`.
```
Comment (thomasten) [inline docs/docs/getting-started/first-steps.md:210]:
```suggestion
    * **iamProfileControlPlane**: The name of an IAM instance profile attached to all control-plane nodes.

        Use the [provided Terraform script](https://github.com/edgelesssys/constellation/blob/main/hack/terraform/aws/iam) to generate the necessary profile. The profile name will be provided as Terraform output value: `control_plane_instance_profile`.

        Alternatively, you can create the AWS profile with a tool of your choice. Use the JSON policy in [main.tf](https://github.com/edgelesssys/constellation/blob/main/hack/terraform/aws/iam/main.tf) in the resource `aws_iam_policy.control_plane_policy`.

    * **iamProfileWorkerNodes**: The name of an IAM instance profile attached to all worker nodes.

        Use the [provided Terraform script](https://github.com/edgelesssys/constellation/blob/main/hack/terraform/aws/iam) to generate the necessary profile. The profile name will be provided as Terraform output value: `worker_nodes_instance_profile`.
```

File: docs/docs/getting-started/install.md [hunk 14]
```diff
@@ -132,6 +132,137 @@ You need the following permissions on this project:
 
 Follow Google's guide on [understanding](https://cloud.google.com/iam/docs/understanding-roles) and [assigning roles](https://cloud.google.com/iam/docs/granting-changing-revoking-access).
 
+</tabItem>
+<tabItem value="aws" label="AWS">
+
+In the following steps you will use Constellation to perform two tasks: create the infrastructure and create roles for cluster nodes. Both of these actions can be performed by different users, e.g., an administrator to create roles and a DevOps engineer to create the infrastructure. We therefore provide a set of permissions for each.
```
Comment (thomasten) [inline docs/docs/getting-started/install.md:138]:
```suggestion
To set up a Constellation cluster, you need to perform two tasks that require permissions: create the infrastructure and create roles for cluster nodes. Both of these actions can be performed by different users, e.g., an administrator to create roles and a DevOps engineer to create the infrastructure.
```

File: docs/docs/overview/clouds.md [hunk 15]
```diff
@@ -32,7 +32,7 @@ The [CVMs available in GCP](https://cloud.google.com/compute/confidential-vm/doc
 
 ## Amazon Web Services (AWS)
 
-AWS currently doesn't offer CVMs. AWS proprietary Nitro Enclaves offer some related features, but [are explicitly not designed to keep AWS itself out](https://aws.amazon.com/blogs/security/confidential-computing-an-aws-perspective/). An experimental version of Constellation exists that runs on Nitro Enclaves.
+AWS currently doesn't offer CVMs. AWS proprietary Nitro Enclaves offer some related features but [are explicitly not designed to keep AWS itself out](https://aws.amazon.com/blogs/security/confidential-computing-an-aws-perspective/). Unfortunately, the Nitro Enclaves' architecture isn't designed to run entire Kubernetes nodes inside them. Therefore, Constellation uses regular EC2 instances on the AWS [Nitro](https://aws.amazon.com/ec2/nitro/) platform. Currently, AWS doesn't offer runtime encryption or attestation primitives for the Nitro EC2 VMs. Instead, AWS provides attestation statements via the [NitroTPM], which is a vTPM managed by the Nitro-Hypervisor. Hence, the hypervisor is currently part of Constellation's TCB.
```
Comment (thomasten) [inline docs/docs/overview/clouds.md:35]:
```suggestion
AWS currently doesn't offer CVMs. AWS proprietary Nitro Enclaves offer some related features but [are explicitly not designed to keep AWS itself out](https://aws.amazon.com/blogs/security/confidential-computing-an-aws-perspective/). Besides, they aren't suitable for running entire Kubernetes nodes inside them. Therefore, Constellation uses regular EC2 instances on AWS [Nitro](https://aws.amazon.com/ec2/nitro/) without runtime encryption. Attestation is based on the [NitroTPM], which is a vTPM managed by the Nitro hypervisor. Hence, the hypervisor is currently part of Constellation's TCB.
```

File: docs/docs/workflows/storage.md [hunk 16]
```diff
@@ -110,6 +121,17 @@ The following installation guide gives an overview of how to securely use CSI-ba
...
     ```
 
+</tabItem>
+<tabItem value="aws" label="AWS">
+
+:::caution
+
+Storage isn't yet implemented for AWS. If you require this feature, [let us know](https://github.com/edgelesssys/constellation/issues/new?assignees=&labels=&template=feature_request.md)!
+
+Other CSI drivers that are compatible with Kubernetes on AWS can be used.
```
Comment (thomasten) [inline docs/docs/workflows/storage.md:131]:
```suggestion
Confidential storage isn't yet implemented for AWS. If you require this feature, [let us know](https://github.com/edgelesssys/constellation/issues/new?assignees=&labels=&template=feature_request.md)!

You may use other (non-confidential) CSI drivers that are compatible with Kubernetes on AWS.
```

File: docs/docs/workflows/storage.md [hunk 17]
```diff
@@ -267,5 +289,16 @@ The examples above are defined to be automatically set as the default storage cl
...
       ```
 
+</tabItem>
+<tabItem value="aws" label="AWS">
+
+:::caution
+
+Storage isn't yet implemented for AWS. If you require this feature, [let us know](https://github.com/edgelesssys/constellation/issues/new?assignees=&labels=&template=feature_request.md)!
+
+Other CSI drivers that are compatible with Kubernetes on AWS can be used.
```
Comment (thomasten) [inline docs/docs/workflows/storage.md:299]:
```suggestion
Confidential storage isn't yet implemented for AWS. If you require this feature, [let us know](https://github.com/edgelesssys/constellation/issues/new?assignees=&labels=&template=feature_request.md)!

You may use other (non-confidential) CSI drivers that are compatible with Kubernetes on AWS.
```

File: docs/docs/getting-started/first-steps.md [hunk 18]
```diff
@@ -164,6 +171,46 @@ If you don't have a cloud subscription, check out [MiniConstellation](first-step
...
+
+        | AMI | Region |
+        | - | - |
+        | `ami-0xxxxxxxxxxxxxxxx` | `eu-central-1` |
+        | `ami-0xxxxxxxxxxxxxxxx` | `us-east-2` |
+        | `ami-0xxxxxxxxxxxxxxxx` | `ap-south-1` |
+
+    * **iamProfileControlPlane**: The name of a profile attached to all control plane nodes.
+
+        Use the [provided Terraform script](https://github.com/edgelesssys/constellation/blob/main/hack/terraform/aws/iam) to generate the necessary profile. It will be provided as Terraform output variable: `control_plane_instance_profile`.
```
Comment (3u13r) [inline docs/docs/getting-started/first-steps.md:204]: Since there is currently a [discussion on where to put the script](https://github.com/edgelesssys/constellation/pull/478), I'd suggest to pin a commit in the link. Even more so, because when we implement `constellation iam create` we will probably move those files again. 

I'm also fine with linking against the release branch, as the place _should_ be stable.

**Review by daniel-weisse — APPROVED** [review]

File: docs/docs/getting-started/install.md [hunk 19]
```diff
@@ -132,6 +132,137 @@ You need the following permissions on this project:
...
+                "iam:DeleteRole"
+            ],
+            "Resource": "*"
+        }
+    ]
+}
+```
+
+To create the infrastructure, you can either use a predefined role from Amazon,
+such as `AmazonEC2FullAccess`, or use the following minimal set of permissions:
```
Comment (3u13r) [inline docs/docs/getting-started/install.md:178]: `AmazonEC2FullAccess` alone is not sufficient since i.e. `logs:*` permissions form below are missing.

**Review by 3u13r — CHANGES_REQUESTED** [review]

**Review by 3u13r — APPROVED** [review]
LGTM
